File: micro-game-lobby/src/topics.rs
#[derive(Debug, Clone)]
pub struct StreamTopics {
    pub find_public_game: String,
    pub create_game: String,
    pub join_private_game: String,
    pub wait_for_opponent: String,
    pub game_ready: String,
    pub private_game_rejected: String,
    pub session_disconnected: String,
}

const DEFAULT_FIND_PUBLIC_GAME: &str = "bugout-find-public-game-cmd";
const DEFAULT_CREATE_GAME: &str = "bugout-create-game-cmd";
const DEFAULT_WAIT_FOR_OPPONENT: &str = "bugout-wait-for-opponent-ev";
const DEFAULT_GAME_READY: &str = "bugout-game-ready-ev";
const DEFAULT_PRIVATE_GAME_REJECTED: &str = "bugout-private-game-rejected-ev";
const DEFAULT_JOIN_PRIVATE_GAME: &str = "bugout-join-private-game-cmd";
const DEFAULT_SESSION_DISCONNECTED: &str = "bugout-session-disconnected-ev";

impl Default for StreamTopics {
    fn default() -> Self {
        StreamTopics {
            find_public_game: DEFAULT_FIND_PUBLIC_GAME.to_string(),
            create_game: DEFAULT_CREATE_GAME.to_string(),
            join_private_game: DEFAULT_JOIN_PRIVATE_GAME.to_string(),
            wait_for_opponent: DEFAULT_WAIT_FOR_OPPONENT.to_string(),
            game_ready: DEFAULT_GAME_READY.to_string(),
            private_game_rejected: DEFAULT_PRIVATE_GAME_REJECTED.to_string(),
            session_disconnected: DEFAULT_SESSION_DISCONNECTED.to_string(),
        }
    }
}

File: kafka-reference-impl/game-lobby/src/main/kotlin/Topics.kt
object Topics {
    const val FIND_PUBLIC_GAME = "bugout-find-public-game-cmd"
    const val CREATE_GAME = "bugout-create-game-cmd"
    const val WAIT_FOR_OPPONENT = "bugout-wait-for-opponent-ev"
    const val GAME_READY = "bugout-game-ready-ev"
    const val PRIVATE_GAME_REJECTED = "bugout-private-game-rejected-ev"
    const val JOIN_PRIVATE_GAME = "bugout-join-private-game-cmd"

    const val GAME_LOBBY_COMMANDS = "bugout-game-lobby-commands"
    const val GAME_LOBBY_CHANGELOG = "bugout-game-lobby"

    const val GAME_STATES_CHANGELOG = "bugout-game-states"
    
    const val SESSION_DISCONNECTED = "bugout-session-disconnected-ev"

    // kv stores
    const val GAME_LOBBY_STORE_LOCAL = "bugout-game-lobby-store-local"
    const val GAME_LOBBY_STORE_GLOBAL = "bugout-game-lobby-store-global"


    val all = arrayOf(FIND_PUBLIC_GAME, CREATE_GAME, WAIT_FOR_OPPONENT,
        GAME_READY, PRIVATE_GAME_REJECTED, JOIN_PRIVATE_GAME,
        GAME_LOBBY_COMMANDS, GAME_LOBBY_CHANGELOG, GAME_STATES_CHANGELOG,
        SESSION_DISCONNECTED)
}
